src/tests/ftest/deployment/network_failure.py

```diff
@@ -0,0 +1,399 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import os
+import time
+
+from ior_test_base import IorTestBase
+from ior_utils import IorCommand
+from general_utils import report_errors, run_pcmd
+from command_utils_base import CommandFailure
+from job_manager_utils import get_job_manager
+from network_utils import update_network_interface
+
+
+class NetworkFailureTest(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description: Verify network failure is properly handled and recovered.
+
+    :avocado: recursive
+    """
+    def __init__(self, *args, **kwargs):
+        """Store the info used during the test and the tearDown."""
+        super().__init__(*args, **kwargs)
+        self.network_down_host = None
+        self.interface = None
+        self.test_env = self.params.get("test_environment", "/run/*")
+
+    def tearDown(self):
+        """Bring ib0 back up in case the test crashed in the middle."""
+        if self.test_env == "ci":
+            self.log.debug("Call ip link set before tearDown.")
+            if self.network_down_host:
+                update_network_interface(
+                    interface=self.interface, state="up", host=self.network_down_host)
+
+        super().tearDown()
+
+    def run_ior_report_error(self, results, job_num, file_name, pool, container,
```

Comment:
Minor: Seeing this method under target_failure.py as well. I think it is worth moving into some utils file.

Reply:
Right. They're slightly different, so I'll work on it in the subsequent PR.